src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_ArrayCreationAndInitializer.vb

```diff
@@ -0,0 +1,435 @@
+          ILiteralExpression (Text: 2) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 2)
+    IArrayInitializer (OperationKind.ArrayInitializer)
+      IArrayInitializer (OperationKind.ArrayInitializer)
+        IArrayInitializer (OperationKind.ArrayInitializer)
+          ILiteralExpression (Text: 3) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 3)
+          ILiteralExpression (Text: 4) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 4)
+]]>.Value
+
+            VerifyOperationTreeForTest(Of CollectionInitializerSyntax)(source, expectedOperationTree)
+        End Sub
+
+        <Fact, WorkItem(17596, "https://github.com/dotnet/roslyn/issues/17596")>
+        Public Sub ArrayCreationErrorCase_MissingDimension()
+            Dim source = <![CDATA[
+Class C
+    Public Sub F()
+        Dim a = New String(1,) {}'BIND:"New String(1,) {}"
+    End Sub
+End Class
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IArrayCreationExpression (Dimension sizes: 2, Element Type: System.String) (OperationKind.ArrayCreationExpression, Type: System.String(,), IsInvalid)
+  IBinaryOperatorExpression (BinaryOperationKind.IntegerAdd) (OperationKind.BinaryOperatorExpression, Type: System.Int32, Constant: 2)
+    Left: ILiteralExpression (Text: 1) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+    Right: ILiteralExpression (Text: 1) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+  IBinaryOperatorExpression (BinaryOperationKind.Invalid) (OperationKind.BinaryOperatorExpression, Type: System.Int32, IsInvalid)
+    Left: IInvalidExpression (OperationKind.InvalidExpression, Type: ?, IsInvalid)
+    Right: ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+  IArrayInitializer (OperationKind.ArrayInitializer)
+]]>.Value
+
+            VerifyOperationTreeForTest(Of ArrayCreationExpressionSyntax)(source, expectedOperationTree)
+        End Sub
+
+        <Fact, WorkItem(17596, "https://github.com/dotnet/roslyn/issues/17596")>
+        Public Sub ArrayCreationErrorCase_InvalidInitializer()
+            Dim source = <![CDATA[
+Class C
+    Public Sub F()
+        Dim a = New C() {1}'BIND:"New C() {1}"
+    End Sub
+End Class
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IArrayCreationExpression (Dimension sizes: 1, Element Type: C) (OperationKind.ArrayCreationExpression, Type: C(), IsInvalid)
+  ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+  IArrayInitializer (OperationKind.ArrayInitializer, IsInvalid)
+    IConversionExpression (ConversionKind.Basic, Implicit) (OperationKind.ConversionExpression, Type: C, IsInvalid)
+      ILiteralExpression (Text: 1) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+]]>.Value
+
+            VerifyOperationTreeForTest(Of ArrayCreationExpressionSyntax)(source, expectedOperationTree)
+        End Sub
+    End Class
```

Comment:
>End Class [](start = 4, length = 9)

Consider testing cases like these:
```
        x = New Integer(2) {1, 2}
        x = New Integer(x(0)) {1, 2}

        Dim y = New Integer(x(0), x(1)) {}
        y = New Integer(1, 1) {}
        y = New Integer(1, 1) {{}}
        y = New Integer(1, 1) {{}, {}}
        y = New Integer(1, 1) {{1, 2}}
        y = New Integer(1, 1) {{1, 2}, {}}
        y = New Integer(1, 1) {{1, 2}, {1, 2}}
```

Reply:
Are we testing these scenarios?

---
In reply to: [108760949](https://github.com/dotnet/roslyn/pull/18154#discussion_r108760949) [](ancestors = 108760949)

Reply:
Done - sorry I had missed this comment.